x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.ts

```diff
@@ -5,16 +5,90 @@
  */
 
 import { SavedObjectsClientContract } from 'kibana/server';
+import { Field } from '../../fields/field';
+import { loadFieldsFromYaml } from '../../elasticsearch/template/install';
+import * as Registry from '../../../registry';
 
-export async function installIndexPattern(
-  name: string,
+interface IndexPatternField extends Field {
+  searchable: boolean;
+  aggregatable: boolean;
+  readFromDocValues: boolean;
+}
+export async function installIndexPatterns(
+  pkgkey: string,
   savedObjectsClient: SavedObjectsClientContract
 ) {
-  await savedObjectsClient.create('index-pattern', getData(name));
+  const registryPackageInfo = await Registry.fetchInfo(pkgkey);
+  const fields = await loadFieldsFromYaml(registryPackageInfo);
+  const kibanaIndexPatternFields = makeKibanaIndexPatternFields(fields);
+  // TODO: separate out logs and metrics patterns
+  await savedObjectsClient.create(
+    'index-pattern',
+    getData('logs-metrics', kibanaIndexPatternFields)
+  );
 }
 
-function getData(name: string) {
+function getData(name: string, fields: IndexPatternField[]) {
   return {
     title: name + '-*',
+    fields: JSON.stringify(fields),
   };
 }
+
+const makeKibanaIndexPatternFields = (fields: Field[]): IndexPatternField[] => {
+  const flattenedFields = flattenFields(fields);
+  const transformedFields = transformFields(flattenedFields);
+  return transformedFields;
+};
+
+const transformFields = (fields: Field[]): IndexPatternField[] => {
+  return fields.map(item => {
+    const newItem = { ...item };
+
+    // map this type to field type
+    if (typeMap[item.type]) {
+      newItem.type = typeMap[item.type];
+    }
+
+    // add some temp values
+    return {
+      searchable: false,
+      aggregatable: false,
+      readFromDocValues: true,
+      ...newItem,
+    };
+  });
+};
+
+const flattenFields = (fields: Field[]): Field[] =>
```

Comment:
This method is great, I'm pretty sure we will need it in other places too. Would be great to get some unit tests for it to tests also some edge cases.